src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int FIGURE_RANDOM_PARAMETR = 20;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomNumber = random.nextInt(FIGURE_COUNT);
+
+        switch (randomNumber) {
+            case 0:
+                int radius = FIGURE_RANDOM_PARAMETR;
+                String colorCircle = colorSupplier.getRandomColor();
```

Comment:
when you create each shape, you call the same method - getRandomColor. Is it better to call this method in the getRandomFigure method and pass the result in the parameters of each method? So you will get rid of duplication of code